src/MSIdentityScaffolding/Microsoft.DotNet.MSIdentity/MicrosoftIdentityPlatform/MicrosoftIdentityPlatformApplicationManager.cs

```diff
@@ -250,21 +295,31 @@ await graphServiceClient.Oauth2PermissionGrants
         /// <param name="createdApplication"></param>
         /// <param name="effectiveApplicationParameters"></param>
         /// <returns></returns>
-        internal static async Task AddPasswordCredentials(GraphServiceClient graphServiceClient, string applicatonId, ApplicationParameters effectiveApplicationParameters)
+        internal static async Task<string> AddPasswordCredentials(GraphServiceClient graphServiceClient, string applicatonId, ApplicationParameters effectiveApplicationParameters)
         {
+            string? password = string.Empty;
             var passwordCredential = new PasswordCredential
             {
                 DisplayName = "Secret created by dotnet-msidentity tool"
             };
 
             if (!string.IsNullOrEmpty(applicatonId) && graphServiceClient != null)
             {
-                PasswordCredential returnedPasswordCredential = await graphServiceClient.Applications[$"{applicatonId}"]
-                    .AddPassword(passwordCredential)
-                    .Request()
-                    .PostAsync();
-                effectiveApplicationParameters.PasswordCredentials.Add(returnedPasswordCredential.SecretText);
+                try
+                {
+                    PasswordCredential returnedPasswordCredential = await graphServiceClient.Applications[$"{applicatonId}"]
+                        .AddPassword(passwordCredential)
+                        .Request()
+                        .PostAsync();
+                    password = returnedPasswordCredential.SecretText;
+                    effectiveApplicationParameters.PasswordCredentials.Add(password);
+                }
+                catch (ServiceException se)
+                {
+                    Console.WriteLine($"Failed to create password : {se.Error.Message}");
```

Comment:
Should this be a write to stderr?  At the moment, it looks like it is going to stdout.

Reply:
You might want to add an aka.ms link in the error message to explain to the user what to do:
2 things can happen:
- there are already 2 passwords added to the app registration (and therefore you can't add another one)
- app passwords are forbidden in the tenant. This one is new. The private preview was announced yesterday, and I've started a conversation on the impact.

Reply:
switched to stderr.